### PR TITLE
Add internal documentation for thread sampling

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Keep internal doc up to date.

## What

Internal docs for `SIGNALFX_THREAD_SAMPLING_PERIOD` and `SIGNALFX_THREAD_SAMPLING_ENABLED`.

## Tests

N/A

### CODE REVIEW

File: docs/internal-config.md [hunk 1]
```diff
@@ -38,4 +38,6 @@ These settings should be never used by the users.
 ## Unpublished settings
 | `SIGNALFX_METRICS_ENDPOINT_URL` | The URL to where metric exporters send metrics. | `http://localhost:9943/v2/datapoint` |
 | `SIGNALFX_RUNTIME_METRICS_ENABLED` | Enable to activate internal runtime metrics sent to SignalFx. | `false` |
-| `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. The Tracer uses it to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |
\ No newline at end of file
+| `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. The Tracer uses it to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |
```
Comment (theletterf) [inline docs/internal-config.md:41]:
```suggestion
| `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. The tracer uses it to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |
```

File: docs/internal-config.md [hunk 2]
```diff
@@ -38,4 +38,6 @@ These settings should be never used by the users.
...
+| `SIGNALFX_THREAD_SAMPLING_ENABLED` | Enable to activate thread sampling sent to SignalFx. | `false` |
```
Comment (theletterf) [inline docs/internal-config.md:42]:
```suggestion
| `SIGNALFX_THREAD_SAMPLING_ENABLED` | Enable to activate thread sampling. | `false` |
```

File: docs/internal-config.md [hunk 3]
```diff
@@ -38,4 +38,6 @@ These settings should be never used by the users.
...
+| `SIGNALFX_THREAD_SAMPLING_PERIOD` | Sampling period to be used. The Tracer uses it to stop threads for shorto amount of time and fetch all stack traces. This value cannot be lower that `1000` ms. | `1000` |
```
Comment (theletterf) [inline docs/internal-config.md:43]:
```suggestion
| `SIGNALFX_THREAD_SAMPLING_PERIOD` | Sampling period. The tracer uses it to stop threads for a short amount of time and fetch all stack traces. This value cannot be lower that `1000` milliseconds. | `1000` |
```

**Review by theletterf — CHANGES_REQUESTED** [review]

**Review by theletterf — APPROVED** [review]

File: docs/internal-config.md [hunk 4]
```diff
@@ -38,4 +38,6 @@ These settings should be never used by the users.
 ## Unpublished settings
 | `SIGNALFX_METRICS_ENDPOINT_URL` | The URL to where metric exporters send metrics. | `http://localhost:9943/v2/datapoint` |
 | `SIGNALFX_RUNTIME_METRICS_ENABLED` | Enable to activate internal runtime metrics sent to SignalFx. | `false` |
-| `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. The Tracer uses it to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |
\ No newline at end of file
+| `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. The tracer uses it to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |
```
Comment (pellared) [inline docs/internal-config.md:41]:
```suggestion
| `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. It is used to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |
```

I find the term `tracer` confusing. Especially that here we are talking about metrics. Not traces.

File: docs/internal-config.md [hunk 5]
```diff
@@ -38,4 +38,6 @@ These settings should be never used by the users.
...
+| `SIGNALFX_THREAD_SAMPLING_ENABLED` | Enable to activate thread sampling. | `false` |
+| `SIGNALFX_THREAD_SAMPLING_PERIOD` | Sampling period. The tracer uses it to stop threads for a short amount of time and fetch all stack traces. This value cannot be lower that `1000` milliseconds. | `1000` |
```
Comment (pellared) [inline docs/internal-config.md:43]:
```suggestion
| `SIGNALFX_THREAD_SAMPLING_PERIOD` | Sampling period. It definess how often the threads are stopped in order to fetch all stack traces. This value cannot be lower than `1000` milliseconds. | `1000` |
```

I find the term `tracer` confusing. Especially that here we are talking about profiling. Not tracing.

**Review by pellared — APPROVED** [review]

**Review by lachmatt — APPROVED** [review]

**Pull request merged**